CHANGELOG

```diff
@@ -3,6 +3,7 @@ Please view this file on the master branch, on stable branches it's out of date.
 v 7.12.0 (unreleased)
   - Update oauth button logos for Twitter and Google to recommended assets
   - Update browser gem to version 0.8.0 for IE11 support (Stan Hu)
+  - Add "replace" and "upload" functionalities to allows user replace existing file and upload new file into current repository
```

Comment:
To make this mergeable, please:

1. [rebase master and squash your 54 commits to 1](http://doc.gitlab.com/ee/workflow/gitlab_flow.html#squashing-commits-with-rebase)
2. Move this line from 7.12.0 to 7.13.0
3. Remove the extraneous CHANGELOG entries